six/modules/c++/six/source/NITFReadControl.cpp

```diff
@@ -195,33 +195,42 @@ void NITFReadControl::load(nitf::IOInterface& ioInterface,
         {
             if (desid == "XML_DATA_CONTENT")
             {
-                // We better have a user-defined subheader.  For now we
-                // don't actually store off the values, but we do check its
-                // tag and size.  Note that we need to check the subheader
+                // Check whether DES is SICD/SIDD, or something of a different
+                // sort. Note that we need to check the subheader
                 // length first rather than calling tre.getCurrentSize() in
                 // case there is no subheader (in which case
                 // subheader.getSubheaderFields() will throw with a
                 // NITRO-specific message).
                 const sys::Uint64_T
                         subheaderLength(subheader.getSubheaderFieldsLength());
                 if (subheaderLength !=
-                            Constants::DES_USER_DEFINED_SUBHEADER_LENGTH)
+                    Constants::DES_USER_DEFINED_SUBHEADER_LENGTH)
                 {
-                    std::ostringstream ostr;
-                    ostr << "Expected a user-defined subheader size of "
-                         << Constants::DES_USER_DEFINED_SUBHEADER_LENGTH
-                         << " but got " << subheaderLength;
-                    throw except::Exception(Ctxt(ostr.str()));
+                    continue;
                 }
 
                 nitf::TRE tre = subheader.getSubheaderFields();
+                std::string field;
+                try
```

Comment:
Oh I see... after taking a look at the SIDD 1.1 branch's `NITFReadControl::getDataType()`, I'll revise this a little.  Here's the problem... if you don't have your plugin path set, this'll throw which you'll catch and then you'll skip over a real DES.  In that case, you want to throw an exception similar to what the branch does telling them they need to set their plugin path.

I'm not sure now why I didn't use `exists()` on the SIDD 1.1 branch... does this not behave well in these scenarios (TRE without this field, TRE with this field w/o plugin path set, TRE with this field with plugin path set)?

But... if you get an XML_DATA_CONTENT of length 5, even with your plugin path set, you won't have `DESSHSI` in there.  So, need to account for that too.

Since this change is already going to make us depend on them setting their plugin path when reading a SICD, can you fold in the `NITFReadControl::getDataType()` from the SIDD 1.1 branch as part of this work too?  Try to see if you can combine the logic for the two cleanly, at least for the `DESSHSI` piece.